src/server/pfs/server/driver.go

```diff
@@ -3150,8 +3150,8 @@ func (d *driver) getFile(pachClient *client.APIClient, file *pfs.File, offset in
 	if err != nil {
 		return nil, err
 	}
-	// Handle commits to input repos
-	if !provenantOnInput(commitInfo.Provenance) {
+	// Handle commits that use the old hashtree format.
+	if !provenantOnInput(commitInfo.Provenance) || commitInfo.Tree != nil {
```

Comment:
Not sure if it's possible to distinguish here, but this still won't work for reading from unfinished commits, that's not a crucial use case, but it would be nice if it worked.

Reply:
Yeah, I think that is not worth figuring out right now since a lot of this will get thrown away / changed when integrating the new storage layer. That does raise a question about whether looking at the content of open commits should be a thing supported by the new storage layer. I think it could essentially be treated as a merge of the available content (what this means in terms of mid job results seems unclear) at request time, but there are some challenges with this. Are concurrent put files handled by the current implementation when looking at files in an open commit? I imagine, if so, it is just leaning on whatever records exist in etcd during the request.

Reply:
I'm down with punting on this for now.

Concurrent puts are handled by the current implementation, it's just based on whatever's in etcd, so I think just merging whatever's present should work well. We don't presently allow people to looks at results midjob, although I think some people would like to be able to do that, it doesn't seem important enough to be worth the trouble.